design/authenticators/authn_sut/authn_sut_solution_design.md

```diff
@@ -0,0 +1,384 @@
+##### Error Log level
+
+| | Scenario | Log message |
+|--- |-----------------------------------------------------------------------  |----------------------------------------------------------------------------------------|
+| 1  | Authenticator is not enabled (in DB/ENV)                                | Authenticator '{0-authenticator-name}' is not enabled                                                 |
+| 2  | Webservice is not defined in a Conjur policy                            | Webservice '{0-webservice-name}' not found                                          |
+| 3  | User or host is not permitted to authenticate with the webservice               | '{0-role-name}' does not have 'authenticate' privilege on {1-service-name}             |
+| 4  | User or host is not defined in Conjur                                           | '{0-role-name}' wasn't found                                                           |
+| 5  | Login error                               | Login error: ...                                     |
+| 6  | Missing SUT in /authenticate request body | Single use token not provided    |
+| 7  | SUT not found or expired | Single use token is invalid    |
+| 8  | Missing code challenge in /login request headers | Code challenge not provided    |
+| 9  | Invalid code challenge algorithm in /login request headers | Code challenge algorithm not supported    |
+| 10  | Invalid code challenge in /login request headers | Code challenge not valid    |
+| 11  | Missing code verifier in /authenticate request body | Code verifier not provided    |
+| 12  | Code verifier does not match code challenge | Code verifier does not match code challenge    |
+
+###### Debug Log level
+
+|    | Scenario                                              | Log message                                                                            |
+|--- |-------------------------------------------------------|----------------------------------------------------------------------------------------|
+| 1  | After validating SUT | Single use token for {0-role-name} has been validated successfully                |
+
+##### Audit Log Messages
+
+Same as other authenticators - one message for successful authentication, one for failed authentication.
+
+#### Policy
+
+As with other authenticators, the SUT authenticator will be disabled by default.
+It will be enabled by creating an authenticator policy resource in Conjur and
+adding it to the list of enabled authenticators in the Conjur configuration, for
+example in the `CONJUR_AUTHENTICATORS` environment variable.
+
+The authenticator webservice must be declared in Conjur policy:
+
+```yml
+- !policy
+  id: conjur/authn-sut
+  body:
+  - !webservice
+
+  - !group 
+    id: authenticatable
+    annotations:
+      description: Group with permission to authenticate using this authenticator
+
+  - !permit
+    role: !group authenticatable
+    privilege: [ read, authenticate ]
+    resource: !webservice
+```
```

Comment:
I'd recommend we add a setting for defining the single-use token TTL. If not present we can use something reasonable, like 30 seconds.

Reply:
Do you think this is necessary for the initial implementation? I added this under out-of-scope in an effort to keep this implementation small.

Reply:
I'd love to get feedback from @niteshtaneja about if we can defer.

Reply:
yeah I had the same question but then Shlomo said that PSM team confirmed that the default TTL is good enough for them. Given that, it should be okay to defer it from v1 implementation.
Just to clarify though- the default is going to be 8 mins and NOT 30s,  correct?

Reply:
30s is for the single use token which PSM will use to log in. Once PSM logs in, it will have the 8 minute auth token.

Reply:
Sorry I mistook this to be making the auth token configurable. Yes, from what I remember from our conversations with the PSM team, a default 30s single-use token should work for them so we can skip making it configurable for the initial implementation.